File: backend/routes/chef_routes.js
const express = require('express');
const router = express.Router();
const {
  getOrdersForPreparation,
  getOrderDetails,
  startPreparation,
  completePreparation,
  setOrderOnHold,
  getChefOrders,
  evaluateChefPerformance
} = require('../services/chef_service');
const User = require('../models/User');

router.get("/queue", async (req, res) => {
  const orders = await getOrdersForPreparation();
  return res.status(200).json(orders);
});

router.get("/:order_id", async (req, res) => {
  const [response, status] = await getOrderDetails(req.params.order_id);
  return res.status(status).json(response);
});

router.post("/start/:order_id", async (req, res) => {
  const [response, status] = await startPreparation(req.params.order_id);
  return res.status(status).json(response);
});

router.post("/complete/:order_id", async (req, res) => {
  const [response, status] = await completePreparation(req.params.order_id);
  return res.status(status).json(response);
});

router.post("/hold/:order_id", async (req, res) => {
  const { note } = req.body;
  
  if (!note) {
    return res.status(400).json({ error: "note is required" });
  }
  
  const [response, status] = await setOrderOnHold(req.params.order_id, note);
  return res.status(status).json(response);
});

router.get("/dashboard/:chef_id", async (req, res) => {
  const chef = await User.findById(req.params.chef_id);
  if (!chef || !["Chef", "Demoted_Chef"].includes(chef.role)) {
    return res.status(403).json({ error: "Unauthorized. Only chefs can access this." });
  }
  
  const orders = await getChefOrders(req.params.chef_id);
  return res.status(200).json(orders);
});

router.get("/performance/:chef_id", async (req, res) => {
  const chef = await User.findById(req.params.chef_id);
  if (!chef || !["Chef", "Demoted_Chef"].includes(chef.role)) {
    return res.status(403).json({ error: "Unauthorized. Only chefs can access this." });
  }
  
  const [result, status] = await evaluateChefPerformance(req.params.chef_id);
  return res.status(status).json(result);
});

module.exports = router;

File: backend/routes/chef_routes.py
from flask import Blueprint, request, jsonify
from services.chef_service import get_orders_for_preparation, start_preparation, complete_preparation, set_order_on_hold

chef_bp = Blueprint('chefs', __name__)

@chef_bp.get("/queue")
def get_queue():
    orders = get_orders_for_preparation()
    return jsonify(orders), 200

@chef_bp.post("/start/<order_id>")
def start(order_id):
    response, status = start_preparation(order_id)
    return jsonify(response), status

@chef_bp.post("/complete/<order_id>")
def complete(order_id):
    response, status = complete_preparation(order_id)
    return jsonify(response), status

@chef_bp.post("/hold/<order_id>")
def hold(order_id):
    data = request.get_json()
    note = data.get("note")

    if not note:
        return jsonify({"error": "note is required"}), 400

    response, status = set_order_on_hold(order_id, note)
    return jsonify(response), status